graylog2-server/src/main/java/org/graylog2/contentpacks/facades/dashboardV1/DashboardWidgetConverter.java

```diff
@@ -0,0 +1,465 @@
+import org.graylog.plugins.views.search.views.widgets.aggregation.BarVisualizationConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.Interpolation;
+import org.graylog.plugins.views.search.views.widgets.aggregation.LineVisualizationConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.NumberVisualizationConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.PivotDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.SeriesConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.SeriesDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.TimeHistogramConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.ValueConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.VisualizationConfigDTO;
+import org.graylog.plugins.views.search.views.widgets.aggregation.sort.PivotSortConfig;
+import org.graylog.plugins.views.search.views.widgets.aggregation.sort.SortConfigDTO;
+import org.graylog2.contentpacks.model.entities.DashboardWidgetEntity;
+import org.graylog2.contentpacks.model.entities.WidgetConfig;
+import org.graylog2.contentpacks.model.entities.WidgetEntity;
+import org.graylog2.contentpacks.model.entities.references.ValueReference;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class DashboardWidgetConverter {
+    private Map<String, ValueReference> parameters;
+    private RandomUUIDProvider randomUUIDProvider;
+    private DashboardWidgetEntity dashboardWidgetEntity;
+    private WidgetConfig config;
+
+    public DashboardWidgetConverter(DashboardWidgetEntity dashboardWidgetEntity,
+                                    Map<String, ValueReference> parameters,
+                                    RandomUUIDProvider randomUUIDProvider) {
+        this.dashboardWidgetEntity = dashboardWidgetEntity;
+        this.parameters = parameters;
+        this.randomUUIDProvider = randomUUIDProvider;
+        this.config = new WidgetConfig(dashboardWidgetEntity.configuration(), parameters);
+    }
+
+    public static List<WidgetEntity> convert(DashboardWidgetEntity dashboardWidgetEntity,
+                                             Map<String, ValueReference> parameters,
+                                             RandomUUIDProvider randomUUIDProvider) {
+        final DashboardWidgetConverter converter = new DashboardWidgetConverter(
+                dashboardWidgetEntity, parameters, randomUUIDProvider);
+        return converter.convert();
+    }
+
+    public List<WidgetEntity> convert() {
+        final String type = dashboardWidgetEntity.type().asString(parameters);
+
+        try {
+            switch (type) {
```

Comment:
I think we should use `type.toUpperCase()` as old dashboard widgets can have their type in lower case. Was different for two different content packs with different dashboards I created on graylog.torch.sh.